src/tests/functional/plugin/gpu/single_layer_tests/dynamic/reduce.cpp

```diff
@@ -0,0 +1,265 @@
+
+            return std::make_shared<Function>(results, params, "ReduceLayerGPUTest");
+        };
+
+        function = makeFunction(params, reduce);
+    }
+};
+
+TEST_P(ReduceLayerGPUTest, CompareWithRefs) {
+   SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+   run();
+}
+
+namespace {
+
+const std::vector<bool> keepDims = {
+    true,
+    false,
+};
+
+const std::vector<ElementType> floatPrecisions = {
+    ElementType::f32,
+    ElementType::f16,
+};
+
+const std::vector<ElementType> floatIntPrecisions = {
+    ElementType::f32,
+    ElementType::f16,
+    ElementType::i32,
+};
+
+
+
+namespace Reduce {
+
+const ReduceInput dyn1d = {
+    {
+        { {-1}, {{4}, {5}, {3}} }
+    },
+    {0}
+};
+
+const ReduceInput dyn2d = {
+    {
+        { {-1, -1}, {{4, 4}, {5, 5}, {6, 1}} }
+    },
+    {1}
+};
+
+const ReduceInput dyn3d = {
+    {
+        { {-1, -1, -1}, {{4, 4, 4}, {5, 1, 5}, {6, 6, 1}} }
+    },
+    {0, 2}
+};
+
+const ReduceInput dyn4d = {
+    {
+        { {-1, -1, -1, -1}, {{2, 3, 4, 5}, {5, 5, 5, 1}, {3, 4, 5, 6}} }
+    },
+    {1, 2}
+};
+
+const ReduceInput dyn5d = {
+    {
+        { {-1, -1, -1, -1, -1}, {{4, 4, 4, 4, 4}, {5, 5, 5, 1, 5}, {1, 4, 5, 6, 7}} }
+    },
+    {2, 3}
+};
+
+const ReduceInput dyn6d = {
+    {
+        { {-1, -1, -1, -1, -1, -1}, {{2, 3, 4, 5, 6, 7}, {5, 5, 5, 1, 5, 5}, {1, 4, 5, 6, 7, 8}} }
+    },
+    {1}
+};
+
+
+// ================== Reduction int32/float types (Sum, Min, Max, L1) ==================
+const auto reduceSum = ::testing::Combine(
+        ::testing::ValuesIn({dyn1d, dyn5d}),
+        ::testing::ValuesIn(floatIntPrecisions),
+        ::testing::Values(helpers::ReductionType::Sum),
+        ::testing::ValuesIn(keepDims),
+        ::testing::Values(CommonTestUtils::DEVICE_GPU)
+);
+INSTANTIATE_TEST_SUITE_P(smoke_reduce_sum_compareWithRefs_dynamic, ReduceLayerGPUTest, reduceSum, ReduceLayerGPUTest::getTestCaseName);
+
+const auto reduceMin = ::testing::Combine(
+        ::testing::ValuesIn({dyn2d, dyn6d}),
+        ::testing::ValuesIn(floatIntPrecisions),
+        ::testing::Values(helpers::ReductionType::Min),
+        ::testing::ValuesIn(keepDims),
+        ::testing::Values(CommonTestUtils::DEVICE_GPU)
+);
+INSTANTIATE_TEST_SUITE_P(smoke_reduce_min_compareWithRefs_dynamic, ReduceLayerGPUTest, reduceMin, ReduceLayerGPUTest::getTestCaseName);
+
+const auto reduceMax = ::testing::Combine(
+        ::testing::ValuesIn({dyn3d, dyn5d}),
+        ::testing::ValuesIn(floatIntPrecisions),
+        ::testing::Values(helpers::ReductionType::Max),
+        ::testing::ValuesIn(keepDims),
+        ::testing::Values(CommonTestUtils::DEVICE_GPU)
+);
+INSTANTIATE_TEST_SUITE_P(smoke_reduce_max_compareWithRefs_dynamic, ReduceLayerGPUTest, reduceMax, ReduceLayerGPUTest::getTestCaseName);
+
+const auto reduceL1 = ::testing::Combine(
+        ::testing::ValuesIn({dyn4d, dyn6d}),
+        ::testing::ValuesIn(floatIntPrecisions),
+        ::testing::Values(helpers::ReductionType::L1),
```

Comment:
Doesn't ReductionType::L1 have a limitation of data type like L2?

Reply:
L1 is sum of absolute of each elements, which means integer is allowed.
but L2 is sum of root value of square of each elements, which means integer is not allowd.